e2e/tests/rate_limiting/rate_limiting_test.go

```diff
@@ -0,0 +1,249 @@
+
+	chainA, chainB := s.GetChains()
+
+	userA := s.CreateUserOnChainA(ctx, testvalues.StartingTokenAmount)
+	userB := s.CreateUserOnChainB(ctx, testvalues.StartingTokenAmount)
+
+	authority, err := query.ModuleAccountAddress(ctx, govtypes.ModuleName, chainA)
+	s.Require().NoError(err)
+	s.Require().NotNil(authority)
+
+	relayer := s.CreateDefaultPaths(testName)
+	s.StartRelayer(relayer, testName)
+
+	chanAB := s.GetChainAToChainBChannel(testName)
+
+	escrowAddrA := transfertypes.GetEscrowAddress(chanAB.PortID, chanAB.ChannelID)
+	denomA := chainA.Config().Denom
+
+	ibcTokenB := testsuite.GetIBCToken(denomA, chanAB.PortID, chanAB.ChannelID)
+
+	t.Run("No rate limit set: Tranfer succeed", func(_ *testing.T) {
+		userABalBefore, err := s.GetChainANativeBalance(ctx, userA)
+		s.Require().NoError(err)
+		userBBalBefore, err := query.Balance(ctx, chainB, userB.FormattedAddress(), ibcTokenB.IBCDenom())
+		s.Require().NoError(err)
+		s.Require().Zero(userBBalBefore.Int64())
+
+		txResp := s.Transfer(ctx, chainA, userA, chanAB.PortID, chanAB.ChannelID, testvalues.DefaultTransferAmount(denomA), userA.FormattedAddress(), userB.FormattedAddress(), s.GetTimeoutHeight(ctx, chainA), 0, "")
+		s.AssertTxSuccess(txResp)
+
+		packet, err := ibctesting.ParseV1PacketFromEvents(txResp.Events)
+		s.Require().NoError(err)
+		s.Require().NotNil(packet)
+
+		s.Require().Eventually(func() bool {
+			_, err := query.GRPCQuery[chantypes.QueryPacketCommitmentResponse](ctx, chainA, &chantypes.QueryPacketCommitmentRequest{
+				PortId:    chanAB.PortID,
+				ChannelId: chanAB.ChannelID,
+				Sequence:  packet.Sequence,
+			})
+			return err != nil && strings.Contains(err.Error(), "packet commitment hash not found")
+		}, time.Second*70, time.Second)
+
+		userABalAfter, err := s.GetChainANativeBalance(ctx, userA)
+		s.Require().NoError(err)
+
+		// Balanced moved form useA to userB
+		s.Require().Equal(userABalBefore-testvalues.IBCTransferAmount, userABalAfter)
+		escrowBalA, err := query.Balance(ctx, chainA, escrowAddrA.String(), denomA)
+		s.Require().NoError(err)
+		s.Require().Equal(testvalues.IBCTransferAmount, escrowBalA.Int64())
+
+		userBBalAfter, err := query.Balance(ctx, chainB, userB.FormattedAddress(), ibcTokenB.IBCDenom())
+		s.Require().NoError(err)
+		s.Require().Equal(testvalues.IBCTransferAmount, userBBalAfter.Int64())
+	})
+
+	t.Run("Add Outgoing Ratelimit on ChainA", func(_ *testing.T) {
+		resp, err := query.GRPCQuery[ratelimitingtypes.QueryAllRateLimitsResponse](ctx, chainA, &ratelimitingtypes.QueryAllRateLimitsRequest{})
+		s.Require().NoError(err)
+		s.Require().Nil(resp.RateLimits)
+
+		s.addRateLimit(ctx, chainA, userA, denomA, chanAB.ChannelID, authority.String(), 10, 0, 1)
+
+		resp, err = query.GRPCQuery[ratelimitingtypes.QueryAllRateLimitsResponse](ctx, chainA, &ratelimitingtypes.QueryAllRateLimitsRequest{})
+		s.Require().NoError(err)
+		s.Require().Len(resp.RateLimits, 1)
+
+		// Make transfer again and see the flow has been updated.
```

Comment:
Seems like this should be a new section, it doesn't fit into "Add Outgoing RateLimit"